Repository/PhpcrRepository.php

```diff
@@ -119,4 +126,54 @@ protected function buildCollection(array $nodes)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentNode = $this->session->getNode(PathHelper::getParentPath($resolvedPath));
+        if (!$parentNode instanceof NodeInterface) {
+            throw new InvalidArgumentException('No parent node created for '.$path);
+        }
+
+        /** @var PhpcrResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::notNull($resource->getName(), 'The resource needs a name for the creation.');
+            Assert::notNull($resource->getPayloadType(), 'The resource needs a type for the creation');
+
+            $parentNode->addNode($resource->getName(), $resource->getPayloadType());
```

Comment:
I don't thin, that i have persisted the node with that, the main information is missing, right?